src/ModifyTable.jl

```diff
@@ -132,22 +132,22 @@ Generic character of GL2
     k ∈ {1,…, q - 1}
   of degree 1
   with values
-    exp(2π𝑖(2//(q - 1)*i*k))
-    (q^2 - 1)*exp(2π𝑖(2//(q - 1)*i*k))
-    (q^2 + q)*exp(2π𝑖(1//(q - 1)*i*k + 1//(q - 1)*j*k))
-    (q^2 - q)*exp(2π𝑖(1//(q - 1)*i*k))
+    exp(2π𝑖((2*i*k)//(q - 1)))
+    (q^2 - 1)*exp(2π𝑖((2*i*k)//(q - 1)))
+    (q^2 + q)*exp(2π𝑖((i*k + j*k)//(q - 1)))
+    (q^2 - q)*exp(2π𝑖((i*k)//(q - 1)))
 
 ```
 """
-function omega(char::GenericCharacter{T}) where T <: NfPoly
+function omega(char::GenericCharacter)
 	t=parent(char)
 	charid=chartypeid(char)
-	new_char_degree=t.modulusring(1)
-	new_char_values=Vector{CycloFrac{T}}(undef, classtypes(t))
+	new_char_degree=base_ring(t.ring)(1)
+	new_char_values=Vector{GenericCyclo}(undef, classtypes(t))
 	for class in 1:classtypes(t)
-		new_char_values[class]=t.classlength[class]*char[class]//char.degree
+		new_char_values[class]=div(t.classlength[class]*char[class], char.degree)  # TODO can this really be fractional?
```

Comment:
maybe use `divexact` (which you will have to define).

Reply:
I'll do this in a follow up PR together with `omega` for `SimpleGenericCharacter` which is using `div` at the moment too.